In [87]:
# import all the filenames
from allfunctions import *

In [88]:
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


1  def LassoCVModel(filename):

2  def OMPCVModel(filename):

3  def GradientBoostingCVModel(filename):

4  def RandomForestCVModel(filename):

5  def RidgeCVModel(filename):

6  def ElasticNetCVModel(filename):

7  def SVRPolyCVModel(filename):

8  def SVRSigmoidCVModel(filename):

9  def SVRLinearCVModel(filename):

10 def SVRRbfCVModel(filename):

In [89]:
functionList = [OMPCVModel]# [ LassoCVModel, OMPCVModel, GradientBoostingCVModel ] 
''', , 
                 RandomForestCVModel, RidgeCVModel, ElasticNetCVModel, 
                 SVRPolyCVModel, SVRSigmoidCVModel, SVRLinearCVModel, 
                 SVRRbfCVModel ] #OMPCVModel
'''

', , \n                 RandomForestCVModel, RidgeCVModel, ElasticNetCVModel, \n                 SVRPolyCVModel, SVRSigmoidCVModel, SVRLinearCVModel, \n                 SVRRbfCVModel ] #OMPCVModel\n'

In [90]:
def wrapper(functionList = [], filenames = [ 'datasetIPPD.pickle' ]):
    if __name__ == '__main__':
        global model_selection
                
        model_selection = {}
        
        for f in functionList:
            print()
            for filename in filenames:
                out = f(filename)
                newScore = out[filename]['train_rmse_cv_4_picking']

                if f in model_selection:
                    #compare scores and select more accurate model
                    oldScore = model_selection[f]['train_rmse_cv_4_picking']
                    if(newScore < oldScore):
                        print("【【【】】】switching models for %s..." % filename)
                        print("【【【】】】newScore: %f < oldScore: %f" % (newScore, oldScore))
                        print("【【【】】】newModel: %s \n" % f)
                        print(100*'_')
                        model_selection[f] = out[filename]
                else:
                    #no score to compare
                    print("【【】】adding new model for %s \n" % filename)
                    print("【【【】】】newScore: %f " % (newScore))
                    print(100*'_')
                    model_selection[f] = out[filename]
    return model_selection

<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


In [86]:
#
#
#
# START of generating small datasets (X_A,y_A), (X_B,y_B), (X_C,y_C)...

In [79]:
import pickle
with open('datasetIPPD.pickle', 'rb') as handle:
    data = pickle.load(handle)

#extract X_test, Y_test
X_test = data['X_test']
y_test = data['y_test']

# In[16]:

def featureReader(featuresFile):
    #reads featurefiles and projects 3 columns (ppn, sizeGB, Ytime)
    #reads profiledFile (the table from profiling) and reads all 55 columns
    #generates one dataframe with 58 columns and n rows (=size of profiled data from csv)
    import pandas as pd
    from IPython.display import display

    header='ft1 ft2 ft3 ft4 ft5 ft6 ft7 ft8 ft9 ft10 ft11 ft12 ft13 ft14 ft15 ft16 ft17     ft18 ft24 ft25 ft19 ft39 ft20 ft33 ft21 ft35 ft22 ft23 ft34 ft36 ft37 ft38 ft40 ft41     ft42 ft43 ft44 ft45 ft46 ft48 ft47 ft49 ft51 ft50 ft52 ft53 ft54 ft55 ft26 ft27 ft28 ft29 ft30 ft31 ft32'
    
    def value(item):
        return item[item.find('=')+1:]
    
    print("Reading features from %s" % featuresFile)
    df_features = pd.read_table(featuresFile, header=None, delimiter=',',
                       converters={i:value for i in range(55)},
                       names=header.split())
    df_features = df_features.astype(float)
    return df_features


# Signature for A
from IPython.display import display
signatureA = featureReader('./features/cdhitdup_singlecmd_features.txt')

# In[105]:

# Signature for B
signatureB = featureReader('./features/frhit_singlecmd_features.txt')

# ## A=cdhitdup, B=frhit, C=velvetG, D=velvetH

# In[106]:

# Signature for C
signatureC = featureReader('./features/velvetg_singlecmd_features.txt')

# In[107]:

# Signature for D
signatureD = featureReader('./features/velveth_singlecmd_features.txt')

# ## Extractor for extracting features

# In[145]:

cols = ['ft1', 'ft2', 'ft3', 'ft4', 'ft5', 'ft6', 'ft7', 'ft8', 'ft9', 'ft10',
       'ft11', 'ft12', 'ft13', 'ft14', 'ft15', 'ft16', 'ft17', 'ft18', 'ft24',
       'ft25', 'ft19', 'ft39', 'ft20', 'ft33', 'ft21', 'ft35', 'ft22', 'ft23',
       'ft34', 'ft36', 'ft37', 'ft38', 'ft40', 'ft41', 'ft42', 'ft43', 'ft44',
       'ft45', 'ft46', 'ft48', 'ft47', 'ft49', 'ft51', 'ft50', 'ft52', 'ft53',
       'ft54', 'ft55', 'ft26', 'ft27', 'ft28', 'ft29', 'ft30', 'ft31', 'ft32']


# In[273]:

def extractor(signature, X_test, y_test):
    import pandas as pd
    import numpy  as np
    import copy
    
    count = 0
    xt = pd.DataFrame(columns = X_test.columns)
    yt = []
    
    for i in range(len(X_test)):
        if(all(X_test[cols].iloc[[i]].values[0] == signature.iloc[[0]].values[0])):
            #print(X_test[cols].iloc[[i]].values[0])
            #print(signature.iloc[[0]].values[0])
            count = count + 1
            # make deep copies
            xt = xt.append(copy.deepcopy(X_test.iloc[[i]]), ignore_index=True)
            yt.append(copy.deepcopy(y_test[i]))
            
    print("Total matches found : %d " % count)
    
    return xt,np.asarray(yt)


# In[274]:

X_A, y_A = extractor(signatureA, X_test, y_test)
X_A.shape, len(y_A)


# In[275]:

X_B, y_B = extractor(signatureB, X_test, y_test)
X_B.shape, len(y_B)


# In[276]:

X_C, y_C = extractor(signatureC, X_test, y_test)
X_C.shape, len(y_C)


# In[277]:

X_D, y_D = extractor(signatureD, X_test, y_test)
X_D.shape, len(y_D)


# In[278]:

len(X_test)


Reading features from ./features/cdhitdup_singlecmd_features.txt
Reading features from ./features/frhit_singlecmd_features.txt
Reading features from ./features/velvetg_singlecmd_features.txt
Reading features from ./features/velveth_singlecmd_features.txt
Total matches found : 5 
Total matches found : 13 
Total matches found : 23 
Total matches found : 24 


65

In [85]:
#
#
#
# END of generating small datasets (X_A,y_A), (X_B,y_B), (X_C,y_C)...

<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# 1 LassoCVModel--OK

In [5]:
t_lasso = wrapper([LassoCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 20.433597178569439, 'precompute': True}
Lasso(alpha=20.433597178569439, copy_X=True, fit_intercept=True,
   max_iter=10000000.0, normalize=False, positive=False, precompute=True,
   random_state=None, selection='random', tol=1e-06, warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 1025.086098 
____________________________________________________________________________________________________


In [6]:
for key in t_lasso.keys():
    print(key)
    print(t_lasso[key]['model'],'\n\n\n\n',
          t_lasso[key]['train_rmse_cv_4_picking'],
          t_lasso[key]['test_rmse_4_reporting'],
          t_lasso[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function LassoCVModel at 0x7f563e1a1378>
Lasso(alpha=20.433597178569439, copy_X=True, fit_intercept=True,
   max_iter=10000000.0, normalize=False, positive=False, precompute=True,
   random_state=None, selection='random', tol=1e-06, warm_start=False) 



 1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________


In [15]:
names = ['ppn', 'sizeGB', 'ft1', 'ft2', 'ft3', 'ft4', 'ft5', 'ft6', 'ft7', 'ft8',
        'ft9', 'ft10', 'ft11', 'ft12', 'ft13', 'ft14', 'ft15', 'ft16', 'ft17',
        'ft18', 'ft24', 'ft25', 'ft19', 'ft39', 'ft20', 'ft33', 'ft21', 'ft35',
        'ft22', 'ft23', 'ft34', 'ft36', 'ft37', 'ft38', 'ft40', 'ft41', 'ft42',
        'ft43', 'ft44', 'ft45', 'ft46', 'ft48', 'ft47', 'ft49', 'ft51', 'ft50',
        'ft52', 'ft53', 'ft54', 'ft55', 'ft26', 'ft27', 'ft28', 'ft29', 'ft30',
        'ft31', 'ft32']

In [16]:
print(sorted(zip(map(lambda x: x, mo.coef_), 
                 names), reverse=True))

[(-0.0, 'sizeGB'), (-0.0, 'ppn'), (-0.0, 'ft9'), (-0.0, 'ft8'), (0.0, 'ft7'), (-0.0, 'ft6'), (-0.0, 'ft55'), (0.0, 'ft54'), (0.0, 'ft53'), (0.0, 'ft52'), (0.0, 'ft51'), (0.0, 'ft50'), (0.0, 'ft5'), (-0.0, 'ft49'), (0.0, 'ft48'), (0.0, 'ft47'), (-0.0, 'ft46'), (0.0, 'ft45'), (-0.0, 'ft44'), (0.0, 'ft43'), (-0.0, 'ft42'), (0.0, 'ft41'), (-0.0, 'ft40'), (0.0, 'ft4'), (0.0, 'ft39'), (0.0, 'ft38'), (0.0, 'ft37'), (-0.0, 'ft36'), (0.0, 'ft35'), (0.0, 'ft34'), (-0.0, 'ft33'), (0.0, 'ft32'), (-0.0, 'ft31'), (0.0, 'ft30'), (-0.0, 'ft3'), (0.0, 'ft29'), (-0.0, 'ft28'), (-0.0, 'ft27'), (0.0, 'ft26'), (0.0, 'ft25'), (0.0, 'ft24'), (0.0, 'ft23'), (0.0, 'ft22'), (0.0, 'ft21'), (-0.0, 'ft20'), (-0.0, 'ft2'), (-0.0, 'ft19'), (0.0, 'ft18'), (0.0, 'ft17'), (0.0, 'ft16'), (0.0, 'ft15'), (0.0, 'ft14'), (0.0, 'ft13'), (0.0, 'ft12'), (0.0, 'ft11'), (-0.0, 'ft10'), (0.0, 'ft1')]


#>>>>>>>>>>>


# Lasso Final

Lasso(alpha=20.433597178569439, copy_X=True, fit_intercept=True,
   max_iter=10000000.0, normalize=False, positive=False, precompute=True,
   random_state=None, selection='random', tol=1e-06, warm_start=False) 



 1025.08609776 880.826424842 1532.34615385

In [17]:
########################### Test Lasso with Scaling ###########################
# the L1 and L2 regularizers of linear models) assume that all features are 
# centered around 0 and have variance in the same order
t_lasso = wrapper([LassoCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 20.433597178569439, 'precompute': True}
Lasso(alpha=20.433597178569439, copy_X=True, fit_intercept=True,
   max_iter=10000000.0, normalize=False, positive=False, precompute=True,
   random_state=None, selection='random', tol=1e-06, warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 1025.086098 
____________________________________________________________________________________________________


In [18]:
for key in t_lasso.keys():
    print(key)
    print(t_lasso[key]['model'],'\n\n\n\n',
          t_lasso[key]['train_rmse_cv_4_picking'],
          t_lasso[key]['test_rmse_4_reporting'],
          t_lasso[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function LassoCVModel at 0x7f563e1a1378>
Lasso(alpha=20.433597178569439, copy_X=True, fit_intercept=True,
   max_iter=10000000.0, normalize=False, positive=False, precompute=True,
   random_state=None, selection='random', tol=1e-06, warm_start=False) 



 1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________


In [64]:
import pickle 

# get the model
for k in t_lasso.keys():
    print(k)
    mo = t_lasso[k]['model']

# write the model to a file
with open('4C4_lasso_tuned.pickle', 'wb') as handle:
    pickle.dump(mo, handle, protocol=pickle.HIGHEST_PROTOCOL)

<function LassoCVModel at 0x7f563e1a1378>


<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# 2 OMPCVModel --OK

In [19]:
t_OMPC = wrapper([OMPCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictionary. The requested precision might not have been met.

  return_n_iter=True)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictionary. The requested precision might not have been met.

  return_n_iter=True)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictionary. The requested precision might not have been met.

  return_n_iter=True)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictiona


Best parameters set found on development set:

{'n_nonzero_coefs': 3, 'tol': 1e-20}
OrthogonalMatchingPursuit(fit_intercept=True, n_nonzero_coefs=3,
             normalize=True, precompute='auto', tol=1e-20)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 445.063184 
____________________________________________________________________________________________________


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/omp.py:665: RuntimeWarning:  Orthogonal matching pursuit ended prematurely due to linear
dependence in the dictionary. The requested precision might not have been met.

  return_n_iter=True)


In [20]:
for key in t_OMPC.keys():
    print(key)
    omp_mo = t_OMPC[key]['model']
    print(t_OMPC[key]['model'],'\n\n\n',
          t_OMPC[key]['train_rmse_cv_4_picking'],
          t_OMPC[key]['test_rmse_4_reporting'],
          t_OMPC[key]['test_mean_y_4_comparing']
         )

    print(100*'_')

<function OMPCVModel at 0x7f563e1a1400>
OrthogonalMatchingPursuit(fit_intercept=True, n_nonzero_coefs=3,
             normalize=True, precompute='auto', tol=1e-20) 


 445.063183594 354.963614472 1532.34615385
____________________________________________________________________________________________________


In [65]:
# get the model
for k in t_OMPC.keys():
    print(k)
    mo = t_OMPC[k]['model']

# write the model to a file
with open('4C4_OMPC_tuned.pickle', 'wb') as handle:
    pickle.dump(mo, handle, protocol=pickle.HIGHEST_PROTOCOL)

<function OMPCVModel at 0x7f563e1a1400>


#>>>>>>>>>>>


# OMP Final

OrthogonalMatchingPursuit(fit_intercept=True, n_nonzero_coefs=3,
             normalize=True, precompute='auto', tol=1e-20) 


 445.063183594 354.963614472 1532.34615385

In [21]:
print(omp_mo.n_nonzero_coefs_)
print()
print(sorted(zip(map(lambda x: x, omp_mo.coef_), 
                 names), reverse=True))

3

[(302.3673068598909, 'sizeGB'), (98.0619727553009, 'ft39'), (3.3961833980045406, 'ppn'), (0.49317894769754, 'ft51'), (0.0, 'ft9'), (0.0, 'ft8'), (0.0, 'ft7'), (0.0, 'ft6'), (0.0, 'ft55'), (0.0, 'ft54'), (0.0, 'ft53'), (0.0, 'ft52'), (0.0, 'ft50'), (0.0, 'ft5'), (0.0, 'ft49'), (0.0, 'ft48'), (0.0, 'ft47'), (0.0, 'ft46'), (0.0, 'ft45'), (0.0, 'ft44'), (0.0, 'ft43'), (0.0, 'ft42'), (0.0, 'ft41'), (0.0, 'ft4'), (0.0, 'ft38'), (0.0, 'ft37'), (0.0, 'ft36'), (0.0, 'ft35'), (0.0, 'ft34'), (0.0, 'ft33'), (0.0, 'ft32'), (0.0, 'ft31'), (0.0, 'ft30'), (0.0, 'ft3'), (0.0, 'ft29'), (0.0, 'ft28'), (0.0, 'ft27'), (0.0, 'ft26'), (0.0, 'ft25'), (0.0, 'ft24'), (0.0, 'ft23'), (0.0, 'ft22'), (0.0, 'ft21'), (0.0, 'ft20'), (0.0, 'ft2'), (0.0, 'ft19'), (0.0, 'ft18'), (0.0, 'ft17'), (0.0, 'ft16'), (0.0, 'ft15'), (0.0, 'ft14'), (0.0, 'ft13'), (0.0, 'ft12'), (0.0, 'ft11'), (0.0, 'ft10'), (0.0, 'ft1'), (-4.4203725761317223, 'ft40')]


#>>>>>>>>>>>


# Final Model : OMPC

function OMPCVModel at 0x7f4f23fe96a8>
OrthogonalMatchingPursuit(fit_intercept=True, n_nonzero_coefs=3,
             normalize=True, precompute='auto', tol=1e-20) 


 445.063183594 354.963614472 1532.34615385

# 3 GradientBoostingCVModel -- OK

In [22]:
t_boosting = wrapper([GradientBoostingCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'learning_rate': 0.005, 'loss': 'ls', 'n_estimators': 6000}
GradientBoostingRegressor(alpha=0.9, criterion='friedman_mse', init=None,
             learning_rate=0.005, loss='ls', max_depth=None,
             max_features=None, max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=1,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, presort='auto', random_state=None,
             subsample=1.0, verbose=0, warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 421.478963 
____________________________________________________________________________________________________


In [84]:
for key in t_boosting.keys():
    print(key)
    print(
          t_boosting[key]['model'],'\n\n\n',
          t_boosting[key]['train_rmse_cv_4_picking'],
          t_boosting[key]['test_rmse_4_reporting'],
          t_boosting[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function GradientBoostingCVModel at 0x7f563e1a1488>
GradientBoostingRegressor(alpha=0.9, criterion='friedman_mse', init=None,
             learning_rate=0.005, loss='ls', max_depth=None,
             max_features=None, max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=1,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, presort='auto', random_state=None,
             subsample=1.0, verbose=0, warm_start=False) 


 421.478963461 342.771761754 1532.34615385
____________________________________________________________________________________________________


In [83]:
model = t_boosting[key]['model']

#A
which = 'A'
X = X_A
y = y_A
######################################################################
reporting_testscore = rmse_scorer(model, X, y)

print('%s   || model : %s' % (which,str(model)[:10]))
print('%s   || test_rmse_reporting: %f' % (which,reporting_testscore))
print('%s   || test_mean_y_comparing: %f' % (which, y.mean()))
print()

#B
which = 'B'
X = X_B
y = y_B
######################################################################
reporting_testscore = rmse_scorer(model, X, y)

print('%s   || model : %s' % (which,str(model)[:10]))
print('%s   || test_rmse_reporting: %f' % (which,reporting_testscore))
print('%s   || test_mean_y_comparing: %f' % (which, y.mean()))
print()

#C
which = 'C'
X = X_C
y = y_C
######################################################################
reporting_testscore = rmse_scorer(model, X, y)

print('%s   || model : %s' % (which,str(model)[:10]))
print('%s   || test_rmse_reporting: %f' % (which,reporting_testscore))
print('%s   || test_mean_y_comparing: %f' % (which, y.mean()))
print()

#D
which = 'D'
X = X_D
y = y_D
######################################################################
reporting_testscore = rmse_scorer(model, X_test, y_test)

print('%s   || model : %s' % (which,str(model)[:10]))
print('%s   || test_rmse_reporting: %f' % (which,reporting_testscore))
print('%s   || test_mean_y_comparing: %f' % (which, y.mean()))
print()

#Full Test Set
which = 'Full Test Set'
X = X_test
y = y_test
######################################################################
reporting_testscore = rmse_scorer(model, X, y)

print('%s   || model : %s' % (which,str(model)[:10]))
print('%s   || test_rmse_reporting: %f' % (which,reporting_testscore))
print('%s   || test_mean_y_comparing: %f' % (which, y.mean()))
print()

A   || model : GradientBo
A   || test_rmse_reporting: 99.034277
A   || test_mean_y_comparing: 117.836000

B   || model : GradientBo
B   || test_rmse_reporting: 2091.025469
B   || test_mean_y_comparing: 2189.093846

C   || model : GradientBo
C   || test_rmse_reporting: 1700.962307
C   || test_mean_y_comparing: 1703.226087

D   || model : GradientBo
D   || test_rmse_reporting: 1591.230073
D   || test_mean_y_comparing: 1307.537500

Full Test Set   || model : GradientBo
Full Test Set   || test_rmse_reporting: 1591.230073
Full Test Set   || test_mean_y_comparing: 1532.346154



In [73]:
# get the model
for k in t_boosting.keys():
    print(k)
    mo = t_boosting[k]['model']

# write the model to a file
with open('4C4_boosting_tuned.pickle', 'wb') as handle:
    pickle.dump(mo, handle)

<function GradientBoostingCVModel at 0x7f563e1a1488>


#>>>>>>>>>>>


# Final Model : Boosting


<function GradientBoostingCVModel at 0x7f4f23fe9730>
GradientBoostingRegressor(alpha=0.9, criterion='friedman_mse', init=None,
             learning_rate=0.005, loss='ls', max_depth=None,
             max_features=None, max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=1,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, presort='auto', random_state=None,
             subsample=1.0, verbose=0, warm_start=False) 


 421.20889837 342.771761754 1532.34615385

<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# 4 Random Forest

In [125]:
t_rforest = wrapper([RandomForestCVModel])


Dataset size read: train 160 and test 161 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'n_estimators': 2000}
RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=6,
           max_features='auto', max_leaf_nodes=None,
           min_impurity_split=1e-07, min_samples_leaf=1,
           min_samples_split=2, min_weight_fraction_leaf=0.0,
           n_estimators=2000, n_jobs=-1, oob_score=True, random_state=None,
           verbose=0, warm_start=False)

Full Test Set: 161


array([ 2087.18,  1333.76,   164.88,   216.71,  2543.24,   692.62,
          97.83,   813.15,    68.75,   323.21,  2441.62,  1623.2 ,
        3450.94,  4234.26,  2684.27,  3344.01,   323.32,  2436.08,
        1141.32,   747.33,   121.17,  1616.34,  1627.26,  2659.86,
         218.82,   164.02,  2661.74,  1879.75,  2828.23,    21.8 ,
        2397.91,   996.42,  2470.53,  1634.05,  1688.66,  1593.28,
         998.83,  1628.58,   823.05,  1083.5 ,   639.5 ,  1632.23,
        2771.25,  2306.07,   216.12,  2271.12,   203.98,  3192.92,
         720.7 ,   801.59,   998.65,  2262.51,  1458.18,  1583.26,
         142.07,  1018.55,  1808.07,  3077.01,  2125.54,   503.45,
        2681.79,  1740.7 ,  1732.64,  1467.65,  1728.17,  1637.05,
        1688.15,  1447.46,  2314.1 ,  2129.21,  2201.04,  1870.14,
        3336.48,   166.8 ,   644.52,  2674.81,  1087.87,   462.06,
        1895.33,  1036.27,  3444.3 ,  1722.37,  2511.62,  2453.84,
        2940.97,  1621.36,  2308.37,  1620.72,   204.44,  1801

array([ 1348.51406618,  1362.87572547,  1348.51406618,  1362.87572547,
        1348.51406618,  1348.51406618,  1348.51406618,  1362.87572547,
        1348.51406618,  1348.51406618,  1348.51406618,  1351.06388118,
        1351.06388118,  1351.06388118,  1348.51406618,  1351.06388118,
        1348.51406618,  1362.87572547,  1362.87572547,  1362.87572547,
        1348.51406618,  1351.06388118,  1362.87572547,  1348.51406618,
        1362.87572547,  1348.51406618,  1351.06388118,  1348.51406618,
        1348.51406618,  1348.51406618,  1351.06388118,  1348.51406618,
        1362.87572547,  1362.87572547,  1348.51406618,  1351.06388118,
        1348.51406618,  1362.87572547,  1362.87572547,  1348.51406618,
        1348.51406618,  1351.06388118,  1348.51406618,  1351.06388118,
        1362.87572547,  1348.51406618,  1348.51406618,  1351.06388118,
        1362.87572547,  1362.87572547,  1348.51406618,  1362.87572547,
        1348.51406618,  1351.06388118,  1348.51406618,  1362.87572547,
      

A: 17


array([ 164.88,   97.83,   68.75,  121.17,  164.02,   21.8 ,  203.98,
        142.07,  166.8 ,  204.44,   21.58,  120.44,   45.94,   97.49,
         45.82,  120.67,   98.42])

array([ 1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618])

B: 48


array([ 1623.2 ,  3450.94,  4234.26,  3344.01,  1616.34,  2661.74,
        2397.91,  1593.28,  1632.23,  2306.07,  3192.92,  1583.26,
        3077.01,  1740.7 ,  1732.64,  1447.46,  2314.1 ,  2201.04,
        3336.48,  1895.33,  3444.3 ,  2511.62,  2940.97,  1621.36,
        2308.37,  1620.72,  1393.36,  1323.62,  1320.38,  2486.13,
        2213.16,  2376.19,  2461.26,  2396.68,  3040.59,  2877.71,
        1861.97,  2024.25,  3359.25,  2762.97,  2210.97,  3049.37,
        2769.22,  2293.62,  2184.34,  2746.73,  3309.8 ,  1423.13])

array([ 1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118,
        1351.06388118,  1351.06388118,  1351.06388118,  1351.06388118])

C: 47


array([ 2087.18,  2543.24,   692.62,   323.21,  2441.62,  2684.27,
         323.32,  2659.86,  1879.75,  2828.23,   996.42,  1688.66,
         998.83,  1083.5 ,   639.5 ,  2771.25,  2271.12,   998.65,
        1458.18,  2681.79,  1728.17,  1688.15,  1870.14,   644.52,
        2674.81,  1087.87,  1722.37,  1001.19,  1857.77,   691.73,
        3143.72,  2451.8 ,  3621.04,  3090.73,  3085.44,  1364.75,
        1470.8 ,  1365.13,   299.12,   297.21,  1472.6 ,  2270.37,
        2084.72,  1080.  ,  1869.08,   697.69,   321.59])

array([ 1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618,  1348.51406618,
        1348.51406618,  1348.51406618,  1348.51406618])

D: 49


array([ 1333.76,   216.71,   813.15,  2436.08,  1141.32,   747.33,
        1627.26,   218.82,  2470.53,  1634.05,  1628.58,   823.05,
         216.12,   720.7 ,   801.59,  2262.51,  1018.55,  1808.07,
        2125.54,   503.45,  1467.65,  1637.05,  2129.21,   462.06,
        1036.27,  2453.84,  1801.74,  1139.61,   814.27,  2106.02,
        1929.21,  1037.18,  1799.43,  2127.71,  1944.16,   748.82,
        2142.12,  1471.89,   214.64,   216.58,   504.63,  1146.09,
         801.4 ,   746.49,  1469.66,   503.73,   460.82,  1134.68,  1467.78])

array([ 1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547,  1362.87572547,  1362.87572547,  1362.87572547,
        1362.87572547])

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 458.549898 
____________________________________________________________________________________________________


In [123]:
for key in t_rforest.keys():
    print(80*'_')
    
    print('Model: %s' % (str(t_rforest[key]['model'])[:20]))
    print('Train RMSE: %f' % (t_rforest[key]['train_rmse_cv_4_picking']))
    
    print('Test RMSE: %f' % (t_rforest[key]['test_rmse_4_reporting']))
    print('Overall Mean: %f' % (t_rforest[key]['test_mean_y_4_comparing']))
    
    print('A...TestA RMSE: %f' % (t_rforest[key]['test_rmse_4_reportingA']))
    print('y_test_A Mean: %f' % (t_rforest[key]['test_mean_y_4_comparingA']))
    
    print('B...TestB RMSE: %f' % (t_rforest[key]['test_rmse_4_reportingB']))
    print('y_test_B Mean: %f' % (t_rforest[key]['test_mean_y_4_comparingB']))
    
    print('C...TestC RMSE: %f' % (t_rforest[key]['test_rmse_4_reportingC']))
    print('y_test_C Mean: %f' % (t_rforest[key]['test_mean_y_4_comparingC']))
    
    print('D...TestD RMSE: %f' % (t_rforest[key]['test_rmse_4_reportingD']))
    print('y_test_D Mean: %f' % (t_rforest[key]['test_mean_y_4_comparingD']))
    
    print(80*'_')

________________________________________________________________________________
Model: RandomForestRegresso
Train RMSE: 444.541537
Test RMSE: 423.563518
Overall Mean: 1590.588075
A...TestA RMSE: 1227.366261
y_test_A Mean: 112.123529
B...TestB RMSE: 1238.462286
y_test_B Mean: 2369.020000
C...TestC RMSE: 952.455539
y_test_C Mean: 1680.930000
D...TestD RMSE: 682.438330
y_test_D Mean: 1254.324694
________________________________________________________________________________


80-20 split:

Model: RandomForestRegresso
Train RMSE: 413.110638
Test RMSE: 340.031393
Overall Mean: 1532.346154
A...TestA RMSE: 834.165178
y_test_A Mean: 117.836000
B...TestB RMSE: 1400.194903
y_test_B Mean: 2189.093846
C...TestC RMSE: 1101.252790
y_test_C Mean: 1703.226087
D...TestD RMSE: 787.724025
y_test_D Mean: 1307.537500

In [67]:
# get the model
for k in t_rforest.keys():
    print(k)
    mo = t_rforest[k]['model']

# write the model to a file
with open('4C4_rforest_tuned.pickle', 'wb') as handle:
    pickle.dump(mo, handle, protocol=pickle.HIGHEST_PROTOCOL)

<function RandomForestCVModel at 0x7f566422c598>


#>>>>>>>>>>>


# Final Model: Random Forest

RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
           max_features='auto', max_leaf_nodes=None,
           min_impurity_split=1e-07, min_samples_leaf=1,
           min_samples_split=2, min_weight_fraction_leaf=0.0,
           n_estimators=4000, n_jobs=-1, oob_score=True, random_state=None,
           verbose=0, warm_start=False) 


 413.344478118 339.417223286 1532.34615385

<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


RidgeCVModel, ElasticNetCVModel, 
                 SVRPolyCVModel, SVRSigmoidCVModel, SVRLinearCVModel, 
                 SVRRbfCVModel ] #OMPCVModel

# RidgeModel

In [26]:
t_ridge = wrapper([RidgeCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 1.0000000000000001e-15}
Ridge(alpha=1.0000000000000001e-15, copy_X=True, fit_intercept=True,
   max_iter=None, normalize=False, random_state=None, solver='auto',
   tol=1e-20)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 445.063184 
____________________________________________________________________________________________________


In [27]:
for key in t_ridge.keys():
    print(key)
    print(
          t_ridge[key]['model'],'\n\n\n',
          t_ridge[key]['train_rmse_cv_4_picking'],
          t_ridge[key]['test_rmse_4_reporting'],
          t_ridge[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function RidgeCVModel at 0x7f563e1a1048>
Ridge(alpha=1.0000000000000001e-15, copy_X=True, fit_intercept=True,
   max_iter=None, normalize=False, random_state=None, solver='auto',
   tol=1e-20) 


 445.063183594 354.963614472 1532.34615385
____________________________________________________________________________________________________


#>>>>>>>>>>>


# Ridge - Final Model

Ridge(alpha=1e-10, copy_X=True, fit_intercept=True, max_iter=None,
   normalize=False, random_state=None, solver='auto', tol=1e-20) 


 570.369986755 478.511724276 1532.34615385

In [28]:
########################### Test Ridge with Scaling ###########################
# the L1 and L2 regularizers of linear models) assume that all features are 
# centered around 0 and have variance in the same order

In [29]:
# Note: Scaling improves Ridge

In [30]:
t_ridge = wrapper([RidgeCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 1.0000000000000001e-15}
Ridge(alpha=1.0000000000000001e-15, copy_X=True, fit_intercept=True,
   max_iter=None, normalize=False, random_state=None, solver='auto',
   tol=1e-20)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 445.063184 
____________________________________________________________________________________________________


In [31]:
for key in t_ridge.keys():
    print(key)
    print(
          t_ridge[key]['model'],'\n\n\n',
          t_ridge[key]['train_rmse_cv_4_picking'],
          t_ridge[key]['test_rmse_4_reporting'],
          t_ridge[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function RidgeCVModel at 0x7f563e1a1048>
Ridge(alpha=1.0000000000000001e-15, copy_X=True, fit_intercept=True,
   max_iter=None, normalize=False, random_state=None, solver='auto',
   tol=1e-20) 


 445.063183594 354.963614472 1532.34615385
____________________________________________________________________________________________________


In [68]:
# get the model
for k in t_ridge.keys():
    print(k)
    mo = t_ridge[k]['model']

# write the model to a file
with open('4C4_ridge_tuned.pickle', 'wb') as handle:
    pickle.dump(mo, handle, protocol=pickle.HIGHEST_PROTOCOL)

<function RidgeCVModel at 0x7f563e1a1048>


<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# Elastic Net

In [32]:
t_elasticNet = wrapper([ElasticNetCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarni


Best parameters set found on development set:

{'l1_ratio': 0.01}
ElasticNet(alpha=3.7926901907322537e-14, copy_X=True, fit_intercept=True,
      l1_ratio=0.01, max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 445.063184 
____________________________________________________________________________________________________


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)


In [33]:
for key in t_elasticNet.keys():
    print(key)
    elasticNet_mo = t_elasticNet[key]['model']
    print(
          t_elasticNet[key]['model'],'\n\n\n',
          t_elasticNet[key]['train_rmse_cv_4_picking'],
          t_elasticNet[key]['test_rmse_4_reporting'],
          t_elasticNet[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function ElasticNetCVModel at 0x7f563e1a10d0>
ElasticNet(alpha=3.7926901907322537e-14, copy_X=True, fit_intercept=True,
      l1_ratio=0.01, max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False) 


 445.063183599 354.963614475 1532.34615385
____________________________________________________________________________________________________


In [34]:
#>>>>>>>>>>>

In [35]:
t_elasticNet = wrapper([ElasticNetCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarni


Best parameters set found on development set:

{'l1_ratio': 0.01}
ElasticNet(alpha=3.7926901907322537e-14, copy_X=True, fit_intercept=True,
      l1_ratio=0.01, max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 445.063184 
____________________________________________________________________________________________________


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)


In [36]:
for key in t_elasticNet.keys():
    print(key)
    elasticNet_mo = t_elasticNet[key]['model']
    print(
          t_elasticNet[key]['model'],'\n\n\n',
          t_elasticNet[key]['train_rmse_cv_4_picking'],
          t_elasticNet[key]['test_rmse_4_reporting'],
          t_elasticNet[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function ElasticNetCVModel at 0x7f563e1a10d0>
ElasticNet(alpha=3.7926901907322537e-14, copy_X=True, fit_intercept=True,
      l1_ratio=0.01, max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False) 


 445.063183599 354.963614475 1532.34615385
____________________________________________________________________________________________________


In [37]:
########################### Test ElasticNet with Scaling ###########################
#
# the L1 and L2 regularizers of linear models) assume that all features are 
# centered around 0 and have variance in the same order

In [38]:
t_elasticNet = wrapper([ElasticNetCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarni


Best parameters set found on development set:

{'l1_ratio': 0.01}
ElasticNet(alpha=3.7926901907322537e-14, copy_X=True, fit_intercept=True,
      l1_ratio=0.01, max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 445.063184 
____________________________________________________________________________________________________


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)


In [39]:
for key in t_elasticNet.keys():
    print(key)
    elasticNet_mo = t_elasticNet[key]['model']
    print(
          t_elasticNet[key]['model'],'\n\n\n',
          t_elasticNet[key]['train_rmse_cv_4_picking'],
          t_elasticNet[key]['test_rmse_4_reporting'],
          t_elasticNet[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function ElasticNetCVModel at 0x7f563e1a10d0>
ElasticNet(alpha=3.7926901907322537e-14, copy_X=True, fit_intercept=True,
      l1_ratio=0.01, max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False) 


 445.063183599 354.963614475 1532.34615385
____________________________________________________________________________________________________


In [40]:
# Let's freeze all optimal parameters - and search for best l1_ratio in [a list] 

t_elasticNet = wrapper([ElasticNetCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarni


Best parameters set found on development set:

{'l1_ratio': 0.01}
ElasticNet(alpha=3.7926901907322537e-14, copy_X=True, fit_intercept=True,
      l1_ratio=0.01, max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 445.063184 
____________________________________________________________________________________________________


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/linear_model/coordinate_descent.py:484: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Fitting data with very small alpha may cause precision problems.
  ConvergenceWarning)


In [41]:
for key in t_elasticNet.keys():
    print(key)
    elasticNet_mo = t_elasticNet[key]['model']
    print(
          t_elasticNet[key]['model'],'\n\n\n',
          t_elasticNet[key]['train_rmse_cv_4_picking'],
          t_elasticNet[key]['test_rmse_4_reporting'],
          t_elasticNet[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function ElasticNetCVModel at 0x7f563e1a10d0>
ElasticNet(alpha=3.7926901907322537e-14, copy_X=True, fit_intercept=True,
      l1_ratio=0.01, max_iter=10000000.0, normalize=False, positive=False,
      precompute=False, random_state=None, selection='cyclic', tol=1e-20,
      warm_start=False) 


 445.063183599 354.963614475 1532.34615385
____________________________________________________________________________________________________


In [69]:
# get the model
for k in t_elasticNet.keys():
    print(k)
    mo = t_elasticNet[k]['model']

# write the model to a file
with open('4C4_elasticNet_tuned.pickle', 'wb') as handle:
    pickle.dump(mo, handle, protocol=pickle.HIGHEST_PROTOCOL)

<function ElasticNetCVModel at 0x7f563e1a10d0>


<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>
<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:><:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>:<:>


# SVR - a - Poly Kernel

In [42]:
t_SVRPolyCVModel = wrapper([SVRPolyCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters... 

Starting grdsurch.fit(X_train, y_train)


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/svm/base.py:220: ConvergenceWarning: Solver terminated early (max_iter=1000000000).  Consider pre-processing your data with StandardScaler or MinMaxScaler.
  % self.max_iter, ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/svm/base.py:220: ConvergenceWarning: Solver terminated early (max_iter=1000000000).  Consider pre-processing your data with StandardScaler or MinMaxScaler.
  % self.max_iter, ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/svm/base.py:220: ConvergenceWarning: Solver terminated early (max_iter=1000000000).  Consider pre-processing your data with StandardScaler or MinMaxScaler.
  % self.max_iter, ConvergenceWarning)
/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/svm/base.py:220: ConvergenceWarning: Solver terminated early (max_iter=1000000000).  Consider pre-processing your data with StandardScaler or MinMaxScaler.


Best parameters set found on development set:

{'C': 1000000000000000.0, 'degree': 6, 'gamma': 1000.0}
SVR(C=1000000000000000.0, cache_size=20480, coef0=0.0, degree=6, epsilon=0.1,
  gamma=1000.0, kernel='poly', max_iter=1000000000.0, shrinking=False,
  tol=0.01, verbose=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 5050183388850412544.000000 
____________________________________________________________________________________________________


/data/home/a1singh/anaconda3/lib/python3.6/site-packages/sklearn/svm/base.py:220: ConvergenceWarning: Solver terminated early (max_iter=1000000000).  Consider pre-processing your data with StandardScaler or MinMaxScaler.
  % self.max_iter, ConvergenceWarning)


In [43]:
t = t_SVRPolyCVModel

for key in t.keys():
    print(key)
    t_mo= t[key]['model']
    print(
          t[key]['model'],'\n\n\n',
          t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function SVRPolyCVModel at 0x7f563e1a1158>
SVR(C=1000000000000000.0, cache_size=20480, coef0=0.0, degree=6, epsilon=0.1,
  gamma=1000.0, kernel='poly', max_iter=1000000000.0, shrinking=False,
  tol=0.01, verbose=False) 


 5.05018338885e+18 8.85063255504e+18 1532.34615385
____________________________________________________________________________________________________


# SVR - Poly Kernel - Final Model

#>>>>>>>>>>>

# SVR - b - Sigmoid Kernel

In [44]:
t_SVRSigmoidCVModel = wrapper([SVRSigmoidCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'C': 1.0000000000000001e-05, 'gamma': 1.0000000000000001e-15}
SVR(C=1.0000000000000001e-05, cache_size=20480, coef0=0.0, degree=3,
  epsilon=0.1, gamma=1.0000000000000001e-15, kernel='sigmoid', max_iter=-1,
  shrinking=False, tol=1e-07, verbose=False)

【【】】adding new model for datasetIPPD.pickle 

【【【】】】newScore: 1032.957231 
____________________________________________________________________________________________________


In [61]:
print()

In [62]:
t = t_SVRSigmoidCVModel

for key in t.keys():
    print(key)
    t_mo= t[key]['model']
    print(
          t[key]['model'],'\n\n\n',
          t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function SVRSigmoidCVModel at 0x7f563e1a11e0>
SVR(C=1.0000000000000001e-05, cache_size=20480, coef0=0.0, degree=3,
  epsilon=0.1, gamma=1.0000000000000001e-15, kernel='sigmoid', max_iter=-1,
  shrinking=False, tol=1e-07, verbose=False) 


 1032.95723148 882.068031346 1532.34615385
____________________________________________________________________________________________________


# SVR - Sigmoid Kernel - Final Model

#>>>>>>>>>>>


# SVR - c - Linear Kernel

In [70]:
t_SVRLinearCVModel = wrapper([SVRLinearCVModel])


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)


Process ForkPoolWorker-425:
Traceback (most recent call last):
Process ForkPoolWorker-432:
Process ForkPoolWorker-420:
  File "/data/home/a1singh/anaconda3/lib/python3.6/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
Process ForkPoolWorker-411:
Process ForkPoolWorker-412:
Process ForkPoolWorker-416:
  File "/data/home/a1singh/anaconda3/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
Traceback (most recent call last):
Process ForkPoolWorker-429:
Process ForkPoolWorker-414:
Process ForkPoolWorker-415:
Process ForkPoolWorker-428:
Traceback (most recent call last):
Process ForkPoolWorker-410:
Process ForkPoolWorker-431:
Process ForkPoolWorker-427:
Traceback (most recent call last):
Process ForkPoolWorker-417:
  File "/data/home/a1singh/anaconda3/lib/python3.6/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
Traceback (most recent call last):
Process ForkPoolWorker-413:
Process ForkPoolWorker-43

TypeError: catching classes that do not inherit from BaseException is not allowed

In [ ]:
t = t_SVRLinearCVModel

for key in t.keys():
    print(key)
    t_mo= t[key]['model']
    print(
          t[key]['model'],'\n\n\n',
          t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

# SVR - Linear Kernel - Final Model

#>>>>>>>>>>>


# SVR - d - rbf Kernel

In [ ]:
t_SVRRbfCVModel = wrapper([SVRRbfCVModel])

In [ ]:
t = t_SVRRbfCVModel

for key in t.keys():
    print(key)
    t_mo= t[key]['model']
    print(
          t[key]['model'],'\n\n\n',
          t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

# SVR - rbf Kernel - Final Model

#>>>>>>>>>>>


# (4C4) Response when four of four modules are chosen as training dataset

'''
### LassoCVModel at 0x7f64208e7c80
1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________
### GradientBoostingCVModel at 0x7f63fa3b07b8
421.108424701 342.356068455 1532.34615385
____________________________________________________________________________________________________
### RandomForestCVModel at 0x7f63fa3b0840
420.621205322 342.402960902 1532.34615385
____________________________________________________________________________________________________
### RidgeCVModel at 0x7f63fa3b08c8
1025.08583073 880.826144692 1532.34615385
____________________________________________________________________________________________________
### ElasticNetCVModel at 0x7f63fa3b0950
1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________
### SVRPolyCVModel at 0x7f63fa3b09d8
1032.95723148 882.068031346 1532.34615385
____________________________________________________________________________________________________
### SVRSigmoidCVModel at 0x7f63fa3b0a60
1080.46665239 1059.84229153 1532.34615385
____________________________________________________________________________________________________
### SVRLinearCVModel at 0x7f63fa3b0ae8
1080.46665239 1059.84229153 1532.34615385
____________________________________________________________________________________________________
### SVRRbfCVModel at 0x7f63fa3b0b70
1080.46665239 1059.84229153 1532.34615385
'''

In [ ]:
import pickle

with open('model_performance_all_modules.pickle', 'wb') as handle:
    pickle.dump(t, handle, protocol=pickle.HIGHEST_PROTOCOL)

In [ ]:
for mem in t.keys():
    print(100*'_')
    print(t[mem])
    print(100*'_')

In [ ]:
t = wrapper()
for mem in t.keys():
    print(100*'_')
    print(t[mem])
    print(100*'_')

# Which features were given most weights by each model ?